sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -247,6 +251,125 @@ def test_lasso_cv_positive_constraint():
     assert min(clf_constrained.coef_) >= 0
 
 
+@pytest.mark.parametrize(
+    "test_model, args",
+    [(Lasso, {"tol": 1e-16, "alpha": 0.1}), (LassoLars, {"alpha": 0.1}),
+     (RidgeClassifier, {"solver": 'sparse_cg', "alpha": 0.1}),
+     (ElasticNet, {"tol": 1e-16, 'l1_ratio': 1, "alpha": 0.1}),
+     (ElasticNet, {"tol": 1e-16, 'l1_ratio': 0, "alpha": 0.1}),
+     (Ridge, {"solver": 'sparse_cg', 'tol': 1e-12, "alpha": 0.1}),
+     (BayesianRidge, {}), (ARDRegression, {}), (OrthogonalMatchingPursuit, {}),
+     (MultiTaskElasticNet, {"tol": 1e-16, 'l1_ratio': 1, "alpha": 0.1}),
+     (MultiTaskElasticNet, {"tol": 1e-16, 'l1_ratio': 0, "alpha": 0.1}),
+     (MultiTaskLasso, {"tol": 1e-16, "alpha": 0.1}),
+     (Lars, {}), (LinearRegression, {}), (LassoLarsIC, {})
+     ])
+def test_model_pipeline_same_as_normalize_true(test_model, args):
+    # Test that linear models (test_model) set with normalize set to True are
+    # doing the same as the same linear model preceeded by StandardScaler
+    # in the pipeline and with normalize set to False
+
+    n_samples, n_features = 100, 2
+    random_state = np.random.RandomState(0)
+    w = random_state.randn(n_features)
+    X = random_state.randn(n_samples, n_features)
+    X += 20  # make features non-zero mean
+
+    y = X.dot(w)
+    # make classes out of regression
+    if 'Classifier' in str(test_model):
+        y[y > np.mean(y)] = -1
+        y[y > 0] = 1
+    if 'MultiTask' in str(test_model):
+        y = np.stack((y, y), axis=1)
+
+    X, X_test, y, y_test = train_test_split(X, y, random_state=42)
+
+    # normalize is True
+    clf_norm = test_model(normalize=True, fit_intercept=True,
+                          **args)
+    clf_norm.fit(X, y)
+    y_pred_norm = clf_norm.predict(X_test)
+
+    if 'alpha' in args:
+        alpha_scaled = args['alpha']
+    if 'Lasso' in str(test_model) and test_model != LassoLarsIC:
+        alpha_scaled = alpha_scaled * np.sqrt(X.shape[0])
+    if 'Ridge' in str(test_model) and 'alpha' in args:
+        alpha_scaled = alpha_scaled * X.shape[0]
+    if 'Elastic' in str(test_model):
+        if args['l1_ratio'] == 1:
+            alpha_scaled = alpha_scaled * np.sqrt(X.shape[0])
+        if args['l1_ratio'] == 0:
+            alpha_scaled = alpha_scaled * X.shape[0]
+
+    if 'alpha' in args:
+        args['alpha'] = alpha_scaled
+
+    clf_pipe = make_pipeline(
+        StandardScaler(),
+        test_model(normalize=False, fit_intercept=True, **args)
+    )
+    clf_pipe.fit(X, y)
+    y_pred_pipe = clf_pipe.predict(X_test)
+
+    assert_array_almost_equal(clf_norm.coef_ * clf_pipe[0].scale_,
+                              clf_pipe[1].coef_, decimal=5)
+    assert clf_pipe[1].intercept_ == pytest.approx(y.mean())
+    assert (clf_norm.intercept_ ==
+            pytest.approx(y.mean() - clf_norm.coef_.dot(X.mean(0))))
```

Comment:
You might want to compute the intercept `y.mean() - clf_norm.coef_.dot(X.mean(0))` and put it in a variable.
It will be easier to read when checking the assert